Jumpscale/builder/apps/BuilderFreeflow.py

```diff
@@ -151,43 +73,64 @@ def sandbox(self, dest_path='/tmp/builder/humhub',create_flist=True, zhub_instan
         :type zhub_instance:str
         '''
         
-        if self._done_check('sandbox') and reset is False:
-            return
-        self.build(reset = reset)
-        dirs = {
-            "/var/www/html/humhub": "/var/www/html/humhub",
-            "/var/lib/mysql/": "/var/lib/mysql/",
-            "/var/log/mysql/": "/var/log/mysql/",
-            "/var/run/mysqld/": "/var/run/mysqld/",
-            "/etc/apache2": "/etc/apache2"
-
-        }
-        self.dirs.update(j.builder.db.mariadb.dirs)
-        bins = ['/usr/bin/mysql','/usr/sbin/apache2']
-        self.bins.extend(j.builder.db.mariadb.bins)
-        for bin_name in bins:
+        self.bins = ['/usr/bin/mysql','/usr/sbin/mysqld','/usr/bin/mysql_install_db','/usr/bin/my_print_defaults','/usr/bin/resolveip','/usr/sbin/apache2','/usr/sbin/apachectl',
+                '/usr/sbin/a2enconf','/usr/sbin/a2enmod','/usr/sbin/a2dismod','/usr/bin/php']
+        for bin_name in self.bins:
             dir_src = self.tools.joinpaths(j.core.dirs.BINDIR, bin_name)
-            dir_dest = j.sal.fs.joinPaths(dest_path, j.core.dirs.BINDIR)
+            dir_dest = j.sal.fs.joinPaths(dest_path, j.core.dirs.BINDIR[1:])
             j.builder.tools.dir_ensure(dir_dest)
             j.sal.fs.copyFile(dir_src, dir_dest)
-        
-        
-        for dir_dest in dirs:
-            dir_dest = j.sal.fs.joinPaths(dest_path, self.tools.path_relative(dir_dest))
-            j.builder.tools.dir_ensure(dir_dest)
-        
         lib_dest = j.sal.fs.joinPaths(dest_path, 'sandbox/lib')
         j.builder.tools.dir_ensure(lib_dest)
-        for bin in bins:
-            dir_src = self.tools.joinpaths(j.core.dirs.BINDIR, bin)
-            j.tools.sandboxer.libs_sandbox(dir_src, lib_dest, exclude_sys_libs=False)
-
-        startup_file = j.sal.fs.joinPaths(j.sal.fs.getDirName(__file__), 'templates', 'freeflow_startup.toml')
+        for bin in self.bins:
+            j.tools.sandboxer.libs_sandbox(bin, lib_dest, exclude_sys_libs=False)
+        self.share = ['/usr/share/mysql']
+        for mysqlshare in self.share:
+            share_dest = j.sal.fs.joinPaths(dest_path, 'sandbox/share/mysql')
+            j.builder.tools.dir_ensure(share_dest)
+            self.tools.copyTree(mysqlshare,share_dest)
+
+        apache_dir = ['/etc/apache2','/usr/lib/apache2','/etc/php','/usr/lib/php','/var/lib/php']
+        for dir in apache_dir:
```

Comment:
why do you do it like that, what is the use of this for loop ? it'd be useful if you made it like this
```python
for dir in apache_dir:
    dest_dir = j.sal.fs.joinPaths(dest_path, 'sandbox', dir)
    j.builder.tools.dir_ensure(dest_dir)
    self.tools.copyTree(dir,dest_dir)
```

Reply:
done